cpp/src/arrow/type.h

```diff
@@ -1600,6 +1611,12 @@ class ARROW_EXPORT Schema : public detail::Fingerprintable,
   bool Equals(const Schema& other, bool check_metadata = false) const;
   bool Equals(const std::shared_ptr<Schema>& other, bool check_metadata = false) const;
 
+  /// \brief Set platform-native endianness in the schema
+  void setNativeEndianness();
```

Comment:
`Schema` is read-only.

How about adding `withNativeEndianness()` or `withEndianness()` like `withMetadata()` instead?

Reply:
I see. Done using `WithNativeEndianness()`